R/test-threads.R

```diff
@@ -0,0 +1,268 @@
+    }
+
+    str_update <- paste0(gsub('[[:digit:]]+', '', str_values[str_loc]), .max_eval)
+    str_line_update <- paste(paste(str_values[!str_loc], collapse = " "), str_update, sep = " ")
+    mod_lines[str_line_loc] <- str_line_update
+    writeLines(mod_lines, mod_path)
+  })
+
+  submit_models(.mods, .mode = .mode, .wait = FALSE, ... = ...)
+
+  .mods
+}
+
+#' Check estimation time for models run with various threads values
+#'
+#' @param .mods a bbi model object or list of model objects. Generally created by `test_threads()`.
+#' @param .return_times character vector denoting which times from `model_summary()` you want to return.
+#'        See details for more information.
+#' @param .wait logical. If TRUE, pass `.mods` to `wait_for_nonmem()` before returning results.
+#' @param ... args passed through to `wait_for_nonmem()`.
+#'
+#' @details
+#' `.return_times` can be any subset of `c("estimation_time", "covariance_time", "cpu_time")`.
+#' Users can also specify `all`, which is the shorthand method for selecting all 3 of those columns.
+#'
+#' @examples
+#' \dontrun{
+#' mods <- test_threads(mod, .threads = c(2, 4))
+#'
+#' If models have not finished:
+#' check_run_times(mods, .wait = TRUE, .time_limit = 300)
+#' check_run_times(mods, .wait = TRUE, .return_times = c("estimation_time", "covariance_time"))
+#'
+#' If models have already finished:
+#' check_run_times(mods, .wait = FALSE)
+#' check_run_times(mods, .wait = FALSE, .return_times = "All")
+#' }
+#' @return A tibble with columns `threads` (number of threads) and `time`
+#'   (elapsed estimation time in seconds for test models).
+#'
+#' @importFrom tidyselect all_of
+#' @importFrom purrr map_dfr
+#'
+#' @export
+check_run_times <- function(
+  .mods,
+  .return_times = "estimation_time",
+  .wait = TRUE,
+  ...
+) {
+
+  if("all" %in% .return_times){
+    .return_times <- c("estimation_time", "covariance_time", "cpu_time")
+  }else{
+    assert_true(all(.return_times %in% c("estimation_time", "covariance_time", "cpu_time")))
+  }
+
+  if(!inherits(.mods, "bbi_nonmem_model")){
+    check_model_object_list(.mods)
+  }else{
+    check_model_object(.mods)
+    .mods <- list(.mods)
+  }
+
+
+  tryCatch({
+    if(.wait) wait_for_nonmem(.mods, ...)
+
+    map_dfr(.mods, ~ {
+      s <- model_summary(.x)
+      threads <- as.numeric(.x$bbi_args$threads)
+      tibble::tibble(
+        model_run = basename(.x$absolute_model_path),
+        threads = threads,
+        estimation_time = s$run_details$estimation_time,
+        covariance_time = s$run_details$covariance_time,
+        cpu_time = s$run_details$cpu_time) %>%
+        select(model_run, threads, all_of(.return_times))
+    })
+  }, error = function(cond){
+    return(NA)
+  }, warning = function(cond){
+    message(cond)
+    message("\nConsider setting/increasing the `time_limit` in `wait_for_nonmem()`. See ?check_run_times for details")
+    return(NA)
+  }
+  )
+}
+
+#' Remove model files associated with the specified tags
```

Comment:
I'm sorry to say this, but I really think we should rename this to `delete_models()`. The "cleanup" name made sense when it was being used for testing, but now that it's exported, I think `delete` is much clearer. Also, all of the other functions spell out "model(s)", so let's keep that consistent.